### ISSUE TITLE
[React] Optimize Filtering

### ISSUE THREAD

**Issue by AmanNegi** [AmanNegi/FreshNest#77]
# Issue

Currently, we re-fetch all the items when we change the filter. However, we sort them in the front end itself. Instead, sort the list of items as we already store them to render the items.

# Files to be changed

`fronted/react/src/pages/shop/presentation/Shop.jsx`
> This is just provided for simplicity of navigation and reference. If required more files can be affected according to needs.

**Comment by bymunajat** [comment]
may i try to help you @AmanNegi

**Comment by AmanNegi** [comment]
Sure, feel free to contribute. ✨🌝
Let me know if you need any help :)

**Comment by AmanNegi** [comment]
> may i try to help you @AmanNegi

You may check the [contributing documentation](https://github.com/AmanNegi/FreshNest/blob/main/docs/CONTRIBUTING.md) for setting up project in your local environment. 😁

**Comment by bymunajat** [comment]
![login](https://github.com/AmanNegi/FreshNest/assets/109524506/80e8e48d-b9ee-4cfa-b93d-b054a13327c5)
how to be able to login?

**Comment by bymunajat** [comment]
> ![login](https://user-images.githubusercontent.com/109524506/252006934-80e8e48d-b9ee-4cfa-b93d-b054a13327c5.jpg) how to be able to login?

@AmanNegi

**Comment by AmanNegi** [comment]
Hey @bymunajat currently there is no signup feature in the interface. You can utilize the API endpoint `/signup` like this:

```js
(POST) http://localhost:3000/api/auth/signup
{ body: { name, email, password, userType, phone } }
```
Start the `backend` and make the following request using any client (POSTMAN, THUNDERCLIENT) and a user will be created with the provided credentials. Let me know if anything is still missing.

**Comment by bymunajat** [comment]
> Hey @bymunajat currently there is no signup feature in the interface. You can utilize the API endpoint `/signup` like this:
> 
> ```js
> (POST) http://localhost:3000/api/auth/signup
> { body: { name, email, password, userType, phone } }
> ```
> 
> Start the `backend` and make the following request using any client (POSTMAN, THUNDERCLIENT) and a user will be created with the provided credentials. Let me know if anything is still missing.

sorry I'm having trouble not being able to log in

**Comment by AmanNegi** [comment]
Did you start the backend server? Note that you will not be working in the production environment, you need to test and develop things out on your local machine.

**Comment by AmanNegi** [comment]
#96 Once this is done, you will be able to add users via user interface itself.

**Comment by AmanNegi** [comment]
@bymunajat  Follow the steps below and you can test directly on the remote website.

1. Clone the project and go into `frontend/react`.
2. Create an `.env` file and add the following line
>  VITE_API_URL="https://agromillets.adaptable.app/api "
3. Run the following commands:
```shell
npm install
npm run dev
```
4. Go to `localhost:5173` and you will enter the login page.
5. Go to SignUp Page and register, you will now be able to access the application.
6. Make changes and test, and create a PR.

> NOTE: Google Login/SignUp will not work until you add the `VITE_GOOGLE_CLIENT_ID` to the `.env` file.

**Comment by AmanNegi** [comment]
Fixed via #105
